Features/bgm 630 deposit halo to x halo

ernestpascual opened the pull request:
## Description of Changes
- Changed all variables of sushibar to halochest
- Change all variables of sushi to halo
- Replaced contract references for useSushiBar() and renamed it to useHaloChest. Placed it under halo-hooks directory
- changed REACT_APP_HALOCHEST_ADDRESS to REACT_APP_HALOCHEST_ADDRESS_KOVAN


[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-21)
[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-35)


### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [x] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/SOOW-XXX-description`
* [x] All files have appropriate file headers and documentation
* [ ] ~~I have written new tests for your core changes, as applicable~~
* [ ] ~~I have successfully ran tests locally~~

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] Code style confirms to eslint/flow rules
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

ernestpascual commented:
- Renamed xhalo to halohalo
- Made addresses dynamic

schystz commented on src/constants/index.ts:
```diff
@@ -72,11 +72,22 @@ export const HALO: ChainTokenMap = {
-// HALO contract addresses
+// HALO Token Addresses
+
+export const HALOAddress: ChainAddressMap = {
```
@ernestpascual consider using all caps to follow the convention used for constants name, I suggest `HALO_TOKEN_ADDRESS`

ernestpascual replied:
gotcha

schystz commented:
@ernestpascual my general feedback when re-engineering is I would advise against renaming the components, hooks, etc., but create a copy and modify that one instead. Leaving the original code as is might be helpful later when debugging (so we have another way to check if wether its a problem with original code or the code we added).

So for example `src/sushi-hooks/useSushiBar.ts → src/halo-hooks/useHaloChest.ts`, we can leave `useSushiBar.ts` alone and make a copy `useHaloChest.ts` instead, and simply use that in the components instead of `useSushiBar`

ernestpascual commented:
Noted on this James, let me bring back the renamed components

ernestpascual commented:
@schystz  updated the PR:
- brought back the renamed components and hooks
- renamed the variable you pointed out

Addtl:
- renamed other halochest to halohalo

schystz commented on src/pages/SushiBar/SushiDepositPanel.tsx:
```diff
@@ -1,5 +1,5 @@
 import React, { useState, useCallback } from 'react'
-import { Pair } from '@sushiswap/sdk'
+import { Currency, Pair } from '@sushiswap/sdk'
```
@ernestpascual I guess the changes here are no longer relevant too right? Shall we revert this file as well?

ernestpascual replied:
https://github.com/sushiswap/sushiswap-interface/blob/5f6374a7afc4a01ef72cec1016fc5f0c5ebd0dfb/src/pages/SushiBar/SushiDepositPanel.tsx#L2

schystz commented on src/pages/SushiBar/XSushiWithdrawlPanel.tsx:
```diff
@@ -8,9 +8,11 @@ import { Input as NumericalInput } from '../../components/NumericalInput'
 import { TYPE } from '../../theme'
 
 import { useActiveWeb3React } from '../../hooks'
+import { useTranslation } from 'react-i18next'
```
@ernestpascual revert this file too?

ernestpascual replied:
https://github.com/sushiswap/sushiswap-interface/blob/5f6374a7afc4a01ef72cec1016fc5f0c5ebd0dfb/src/pages/SushiBar/XSushiWithdrawlPanel.tsx#L11

schystz commented on src/pages/SushiBar/index.tsx:
```diff
@@ -48,7 +48,7 @@ export default function Saave() {
                 <div>
                   <TYPE.white fontSize={14} color={theme.text2} style={{ paddingBottom: '10px' }}>
-                    {`Stake your SUSHI into xSUSHI for ~15% APY. No impermanent loss, no loss of governance rights. Continuously compounding.`}
+                    {`Stake your SUSHI into xSUSHI for ~5% APY. No impermanent loss, no loss of governance rights. Continuously compounding.`}
```
@ernestpascual revert this file too?

Interesting though, the APY is hardcoded?? LOL

ernestpascual replied:
https://github.com/sushiswap/sushiswap-interface/blob/5f6374a7afc4a01ef72cec1016fc5f0c5ebd0dfb/src/pages/SushiBar/index.tsx#L51

schystz commented:
Thanks @ernestpascual . Just a few more minor comments but looks good to me.

ernestpascual commented:
@schystz the files you mentioned are already reverted. i copied over the original sushi files since the halo files are already on their separate folders. Also, good spot at the APY idk they hardcoded it lol

schystz commented on public/locales/en.json:
```diff
@@ -99,7 +99,7 @@
   "poolSummaryHaloEarned": "Unclaimed HALO Earned",
   "totalPoolValue": "Total Pool Value",
-  "tokenCardRewardDescription": "Every HALO token you earn is automatically staked in DESSERT to earn vesting rewards! Press Claim or go to the Vesting tab to manage rewards.",
+  "tokenCardRewardDescription": "Every HALO token you earn is automatically staked in Rainbow Pool to earn vesting rewards! Press Harvest or go to the Vesting tab to manage rewards.",
```
Shouldn't this be the name of the reward token?

```suggestion
  "tokenCardRewardDescription": "Every HALO token you earn is automatically staked in RAINBOW DROP to earn vesting rewards! Press Harvest or go to the Vesting tab to manage rewards.",
```

tracyarciaga replied:
> Shouldn't this be the name of the reward token?

hmmm I think it should still say "staked in Rainbow Pool" because RAINBOW DROP is the token name.
Rainbow Pool is the product name for the Rainbow contract
Rainbow Drop is the token name
RNBW token symbol.

schystz replied:
Yeah, but this is originally "staked in DESSERT", where DESSERT is the token name not the product name. So thought it would make more sense to replace it with the new token name as well - either RAINBOW DROP or Rainbow Drop

tracyarciaga replied:
hmm I think Rainbow Pool makes more sense, even if it was dessert, I think Dessert Pool would have made more sense.. hehe

tracyarciaga replied:
Well we could say automatically staked AS Rainbow Drop though, not IN.. that makes sense too..

tracyarciaga replied:
@schystz i've changed it already

schystz replied:
Jofo was the one who designed the original content so I think better to discuss with him what would be best?

tracyarciaga replied:
Ok will discuss w Jofo, but yes, let's go with rainbow drop for now. I already changed it..

schystz reviewed (commented):
Hi @tracyarciaga , looks good overall. Just 1 minor comment below.

schystz reviewed (approved):
Lgtm after the last couple rounds of feedback 👍

bitcoinbrisbane reviewed (changes requested):
LGTM, just need to fix the listing issues
